<!-- Launch file for the rrt-detector and the assigner -->


<launch>
<arg name="eta" value="1.0"/>
<arg name="Geta" value="15.0"/>
<param name="namespace_init_count" value="1"/>


  <node pkg="rrt_exploration_real_two_cars" type="global_rrt_detector_real_two_cars" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="/robot1/map"/>
  </node>
  
  <node pkg="rrt_exploration_real_two_cars" type="local_rrt_detector_real_two_cars" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="/robot1/map"/>
  <param name="robot_frame" value="robot1/base_footprint"/>
  </node>
 
  
  <node pkg="rrt_exploration_real_two_cars" type="filter.py" name="filter" output="screen">
  <param name="map_topic" value="/robot1/map"/>
  <param name="info_radius" value="1"/> 
  <param name="costmap_clearing_threshold" value="0"/> 
  <param name="goals_topic" value="/detected_points"/>
  <param name="namespace" value="/robot"/> 
  <param name="n_robots" value="1"/>
  <param name="rate" value="100"/>
  </node>
  
  <node pkg="rrt_exploration_real_two_cars" type="assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="/robot1/map"/>
  <param name="global_frame" value="robot1/map"/>
  <param name="info_radius" value="1"/> 
  <param name="info_multiplier" value="3.0"/> 
  <param name="hysteresis_radius" value="3.0"/> 
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="n_robots" value="1"/>
  <param name="namespace" value="/robot"/>
  <param name="delay_after_assignement" value="0.5"/>
  <param name="rate" value="100"/>
  </node>
  
  <!--
<node pkg="mapmerge" type="mapmerge_node" name="map_merge" output="screen" >
	<param name="robot1_x" value="0.0" />
	<param name="robot1_y" value="0.0" />

	<param name="resolution" value="0.05" />
	<param name="map_width" value="600" />
	<param name="map_height" value="600" />
</node>-->


  <!-- run RViz node (visualization) -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrt_exploration_real_two_cars)/rviz/single_car.rviz">
  <remap from="move_base_simple/goal" to="robot1/move_base_simple/goal"/>
  </node>



</launch>
